usage_analytics/analytics_func.py

```diff
@@ -0,0 +1,72 @@
+def get_calling_class(recursive=True):
+    """
+    Traverses the call stack to find and return the class name of the first
+    caller that has a 'self' variable in its frame (i.e., a method call).
+    If no such frame is found, returns 'StandaloneFunction'.
+    """
+
+    import inspect
+
+    # Skip the current frame and the immediate caller (i.e., start from index 1)
+    stack = inspect.stack()[1:]
+
+    outermost_caller = "StandaloneFunction"  # Default return value
+
+    for frame_record in stack:
+        frame = frame_record.frame
+        # Look for 'self' in the frame's local variables
+        self_obj = frame.f_locals.get("self", None)
+        if self_obj is not None:
+            # Return the class name of the 'self' instance
+            outermost_caller = type(self_obj).__name__
+            if not recursive:
+                break
+
+    # If no class context was found, assume it's a standalone function
```

Comment:
remove that comment. It is already apparent from the doc string and the default value of `outmost_caller`.